src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double CHANGES = 1.5;
+    private int sizeThatChanges = 10;
+    private int size = 0;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[sizeThatChanges];
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        items[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        resize();
+        if (index == size) {
+            items[index] = value;
+            size++;
```

Comment:
no need to handle this specifically, arraycopy won't copy anything if size-index == 0